src/WinUsb/WinUsb.cs

```diff
@@ -0,0 +1,167 @@
+        /// </summary>
+        /// <param name="deviceHandle">
+        /// The handle to the device that <see cref="Kernel32.CreateFile(string, Kernel32.ACCESS_MASK, Kernel32.FileShare, Kernel32.SECURITY_ATTRIBUTES*, Kernel32.CreationDisposition, Kernel32.CreateFileFlags, Kernel32.SafeObjectHandle)"/> returned.
+        /// </param>
+        /// <param name="interfaceHandle">
+        /// Receives an opaque handle to the first (default) interface on the device.
+        /// This handle is required by other WinUSB routines that perform operations on the default interface.
+        /// To release the handle, call the <see cref="WinUsb_Free"/> function.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true"/> if the operation succeeds. Otherwise, this routine returns <see langword="false"/>,
+        /// and the caller can retrieve the logged error by calling <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        [DllImport(nameof(WinUsb), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool WinUsb_Initialize(Kernel32.SafeObjectHandle deviceHandle, out SafeUsbHandle interfaceHandle);
+
+        /// <summary>
+        /// Rretrieves information about the specified endpoint and the associated pipe for an interface.
+        /// </summary>
+        /// <param name="interfaceHandle">
+        /// An opaque handle to an interface that contains the endpoint with which the pipe is associated.
+        /// </param>
+        /// <param name="alternateInterfaceNumber">
+        /// A value that specifies the alternate interface to return the information for.
+        /// </param>
+        /// <param name="pipeIndex">
+        /// A value that specifies the pipe to return information about. This value is not the same as the bEndpointAddress field in the endpoint descriptor.
+        /// A <paramref name="pipeIndex"/> value of 0 signifies the first endpoint that is associated with the interface, a value of 1 signifies the second endpoint,
+        /// and so on. <paramref name="pipeIndex"/> must be less than the value in the bNumEndpoints field of the interface descriptor.
+        /// </param>
+        /// <param name="pipeInformation">
+        /// A pointer, on output, to a caller-allocated <see cref="WINUSB_PIPE_INFORMATION"/> structure that contains pipe information.
+        /// </param>
+        /// <returns>
+        /// Returns <see langword="true"/> if the operation succeeds. Otherwise, this routine returns <see langword="false"/>,
+        /// and the caller can retrieve the logged error by calling <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        [DllImport(nameof(WinUsb), SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool WinUsb_QueryPipe(SafeUsbHandle interfaceHandle, byte alternateInterfaceNumber, byte pipeIndex, out WINUSB_PIPE_INFORMATION pipeInformation);
```

Comment:
Please define the last parameter as:

```cs
[Friendly(FriendlyFlags.Out)] WINUSB_PIPE_INFORMATION* pipeInformation
```

And please put each parameter on its own line now that the signature is getting longer.